### PR TITLE
fix: bring hmr back

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#8011** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8011?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/common/infra/src/framework/core/identifier.ts [hunk 1]
```diff
@@ -100,7 +99,7 @@ export function createIdentifier<T>(
 export function createIdentifierFromConstructor<T extends Component>(
   target: Type<T>
 ): Identifier<T> {
-  return createIdentifier<T>(`${target.name}${stableHash(target)}`);
+  return createIdentifier<T>(target.name);
```
Comment (EYHN) [inline packages/common/infra/src/framework/core/identifier.ts]: it work, but not safe

Comment (forehalo) [reply packages/common/infra/src/framework/core/identifier.ts]: why

Comment (EYHN) [reply packages/common/infra/src/framework/core/identifier.ts]: If there are types with the same name, this can overwriting.

Comment (forehalo) [reply packages/common/infra/src/framework/core/identifier.ts]: then use type itself as identifier

**Review by EYHN — DISMISSED** [review]

File: packages/common/infra/src/framework/core/identifier.ts (hunk 1 above)
Comment (EYHN) [reply packages/common/infra/src/framework/core/identifier.ts]: Then hmr is broken, once you use the identifier bound to the type

Comment (EYHN) [reply packages/common/infra/src/framework/core/identifier.ts]: <div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/g3jz87HxbjOJpXV3FPT7/40fa720d-d44e-4396-b4a0-ea89a89be683.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/g3jz87HxbjOJpXV3FPT7/40fa720d-d44e-4396-b4a0-ea89a89be683.mp4">CleanShot 2024-09-04 at 15.20.13.mp4</video>

**Review by darkskygit — APPROVED** [review]

**Review by EYHN — CHANGES_REQUESTED** [review]
not work

**Pull request closed without merging**